support-services/src/main/scala/com/gu/support/redemption/RedemptionCode.scala

```diff
@@ -0,0 +1,13 @@
+package com.gu.support.redemption
+
+object RedemptionCode {
+  def apply(codeAsString: String): Either[String, RedemptionCode] = {
+    // make sure no one can inject anything bad
```

Comment:
actually, the IT test proves that dynamo client can handle all the characters, but even so, it makes sense to parse the redemption code string into the types as early as possible, so we can assume it's correct from the type.